### PR TITLE
wallet: Only fee-bump non-conflicted/non-confirmed txes

### PR DESCRIPTION
**Pull request by maflcko**
This only affects the gui.

Fee-bumping of transactions that are already confirmed or are already conflicted by other transactions should not be offered by the gui.

### CODE REVIEW

**Comment by promag** [comment]
ACK fa4b00f.

**Comment by MarcoFalke** [comment]
Sorry, fixed up typo in commit message.

**Comment by promag** [comment]
re-ACK fa2457e.

It can be a bit heavy, but the check `wallet->IsAllFromMe()` makes a lot of sense, there is no way to bump received transactions.

**Comment by instagibbs** [comment]
Food for thought: Move more checks into `PreconditionChecks` and use that instead: https://github.com/instagibbs/bitcoin/commit/718f05cab5fe632c5dc4e3c689d5e4cd51331089

This makes reasoning about what is supposed to be called easier in the future.

**Comment by promag** [comment]
Restarted travis job due to `AssertionError: Mempool sync failed` in `wallet_backup.py`.

**Comment by MarcoFalke** [comment]
@instagibbs Thx. Your commit seems like a proper fix.

**Comment by promag** [comment]
This should be tested with a big wallet now that `IsMine` is called when the context menu opens.

File: src/wallet/feebumper.cpp [hunk 1]
```diff
@@ -228,6 +234,7 @@ Result CreateTransaction(const CWallet* wallet, const uint256& txid, const CCoin
         }
     }
 
+
```
Comment (promag) [inline src/wallet/feebumper.cpp]: Nit, remove.

**Review by ryanofsky — COMMENTED** [review]
utACK faca18dcf499e36069ce5fcd3e02a5ee86639436

> This should be tested with a big wallet now that IsMine is called when the context menu opens.

I could be misunderstanding the concern, but I'd expect the size of the wallet not to make a big difference if IsMine is only called for the one transaction that's clicked.

**Comment by MarcoFalke** [comment]
@promag Thanks for raising the issue. Though, I confirmed that it is indeed only called when the context menue opens, so it shouldn't be a problem.

**Comment by promag** [comment]
Just saying that the UI will block because of IO and all, and that may be noticeable on big wallets (lots of keys) - reason `AllFromMe` must fetch all spent scriptPubKey and call IsMine for each one.

File: src/wallet/feebumper.cpp [hunk 2]
```diff
@@ -65,16 +65,39 @@ static feebumper::Result PreconditionChecks(const CWallet* wallet, const CWallet
...
         return feebumper::Result::WALLET_ERROR;
     }
+
+    if (!SignalsOptInRBF(*wtx.tx)) {
+        errors.push_back("Transaction is not BIP 125 replaceable");
+        return feebumper::Result::WALLET_ERROR;
+    }
+
+    if (wtx.mapValue.count("replaced_by_txid")) {
+        errors.push_back(strprintf("Cannot bump transaction %s which was already bumped by transaction %s", wtx.GetHash().ToString(), wtx.mapValue.at("replaced_by_txid")));
```
Comment (promag) [inline src/wallet/feebumper.cpp]: Could avoid 2nd lookup, but looks more clean this way.

**Review by promag — COMMENTED** [review]
utACK faca18d.

**Comment by jonasschnelli** [comment]
utACK faca18dcf499e36069ce5fcd3e02a5ee86639436

**Pull request merged**